frontend/app/features/kanban/components/TaskCard.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react";
+import Modal from "../../../components/modal/Modal";
+import { Task } from "../utils/Tasks";
+import EditTaskForm from "./EditTaskForm";
+
+interface TaskCardProps {
+  task: Task;
+  onEdit: () => void;
+}
+
+const TaskCard: React.FC<TaskCardProps> = ({ task }) => {
+  const [isEditing, setIsEditing] = useState(false);
+
+  return (
+    <div className="card mb-3">
+      <header className="card-header">
+        <p className="card-header-title">{task.title}</p>
+      </header>
+      <div className="card-content">
+        <p>{task.description}</p>
+        <p className="has-text-grey-light">
+          Due Date: {task.dueDate ? new Date(task.dueDate).toLocaleDateString() : "No due date"}
+        </p>
+        <p className="has-text-grey">
+          Created By: {task.createdBy} | Assigned To: {task.assignedTo}
+        </p>
+      </div>
+      <footer className="card-footer">
+        <button
+          className="card-footer-item button is-small is-info"
+          onClick={() => setIsEditing(true)}
+        >
+          Edit
+        </button>
+      </footer>
+
+      {isEditing && (
+        <Modal onClose={() => setIsEditing(false)} isOpen={isEditing}>
+          <EditTaskForm
+            task={task}
+            isOpen={isEditing}
+            onClose={() => setIsEditing(false)}
+            closeModal={function (): void {
+              throw new Error("Function not implemented.");
+            }}
```

Comment:
the same error here, like I mentioned before, please carefully check your changes when you receive a comment, bcs the comment applies all related changes, its really disappointing to repeat this over and over again but see no improvement